Automattic/wp-calypso
# Login: Refine keyboard behavior and focus fields login pages

This pull request fixes #14211 and #14212 by refining forms on the `Login` pages in order to:

* Focus the first field on page load
* Focus the `Verification Code` field whenever an inline error happens
* Disable autocompletion on the `Password` and `Verification Code` fields

Regarding mobile devices, it:

* Sets the keyboard layout to lowercase for the `Username or Email Address` and `Password` fields
* Enables the numeric keyboard layout for the `Verification Code` field
 
<img width="365" alt="screenshot" src="https://cloud.githubusercontent.com/assets/594356/26347911/55859688-3fab-11e7-811e-a8bea40f2f84.png">

  
#### Testing instructions

You should test those steps from a mobile device:
 
1. Run `git checkout update/login-pages` and start your server, or open a [live branch](https://calypso.live/log-in?branch=update/login-pages)
2. Open the [`Login` page](http://calypso.localhost:3000/log-in)
3. Check that the `Username or Email Address` field has focus once the page has loaded
4. Check that the keyboard layout is set to lowercase for all fields
5. Check that autocompletion is disabled on the `Password` field
6. Enter credentials for an account with 2FA
7. Check that the `Verification Code` field has focus once the page has loaded
8. Check that autocompletion is disabled on this  field
9. Check that the keyboard layout is set to numeric for the `Verification Code` field
10. Submit a wrong code and check that the `Verification Code` field is given focus again
11. Switch authentication method by clicking any other option provided below the form (if any)
12. Check that the `Verification Code` field has focus again

#### Additional notes
 
@gziolo mentioned [an issue](https://github.com/Automattic/wp-calypso/issues/14212#issuecomment-303296835) with the error message being persisted when switching of authentication method. I think it would be better to address that in a different pull request, especially considering the fact that the success notice is also persisted across `Login` pages (something that shouldn't happen as well).

This pull request also moves the `Return` link at the very bottom of the footer since I don't see why this link should take precedence over the other options (actually I'm not sure what problem we're trying to solve with this link, and I would gladly remove it).
 
#### Reviews
